### PR TITLE
Update to gds, gds-dmt, gds-dtt, gds-gui, gds-monitor-apps, gds-root-extensions…

### PR DESCRIPTION
**Pull request by emaros**
…, gds-swig, gds-utilities: split package

 Because of the size of the GDS project, it was decided to break it down
into several smaller projects. The new set of files are:

* gds updated to version 2.19.1
* gds-dmt a new package starting with version 2.19.1
* gds-dtt a new package starting with version 2.19.0
* gds-gui a new package starting with version 2.19.0
* gds-monitor-apps a new package starting with version 2.19.1
* gds-root-extensions a new package starting with version 2.19.0
* gds-swig a new package starting with version 2.19.0
* gds-utilities a new package starting with version 2.19.0

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.12.6 16G2136
Xcode 9.2 9C40b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by emaros** [comment]
The Travis failure is because of the length of time this project takes to compile. It has never been able to complete within the Travis time limit.
My interpretation is that the Azure pipelines have also timed out during the compilation phase.

**Comment by reneeotten** [comment]
> My interpretation is that the Azure pipelines have also timed out during the compilation phase.

That interpretation does not seem correct (you can click on it and see what failed/happened) as it [fails](https://paste.z0k.xyz/b5efc3e7b46e) to build `gds` due to a checksum issue.

File: science/gds-dtt/Portfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
+
+checksums     rmd160 45f5f06d8190038ba43dede6a6d409467214687d \
+              sha256 8eb3c03845f5fa00bf9b4da104c40cb7773868570ac61c926befffa515f54f60
```
Comment (reneeotten) [inline science/gds-dtt/Portfile:24]: please add the `size` as well

File: science/gds-dtt/Portfile [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+maintainers   {@emaros ligo.org:ed.maros} openmaintainer
+license       GPL
+
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
```
Comment (reneeotten) [inline science/gds-dtt/Portfile:21]: please use `https` if possible

File: science/gds-dmt/Portfile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+maintainers   {@emaros ligo.org:ed.maros} openmaintainer
+license       GPL
+
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
```
Comment (reneeotten) [inline science/gds-dmt/Portfile:20]: please use `https` if possible

File: science/gds-dtt/Portfile [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    configure.env           CXXFLAGS='-std=c++1z'
+
+    depends_lib-append      port:root6 \
+                            port:python${python.version} \
+                            port:py${python.version}-numpy
+
+	require_active_variants root6 python${python.version}
+} else {
+    set python.version      27
+    configure.env           CXXFLAGS='-std=c++11'
```
Comment (reneeotten) [inline science/gds-dtt/Portfile:86]: when adding this flag you likely want to make sure to request a C++11 compatible compiler, so with `compiler.cxx_standard   2011` perhaps?

File: science/gds-dmt/Portfile [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem    1.0
+PortGroup     conflicts_build 1.0
+PortGroup     active_variants 1.1
```
Comment (reneeotten) [inline science/gds-dmt/Portfile:5]: both PotGroups seem unused here...?

File: science/gds-dtt/Portfile [hunk 6]
```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem    1.0
+PortGroup     conflicts_build 1.0
```
Comment (reneeotten) [inline science/gds-dtt/Portfile:4]: unused?

File: science/gds-gui/Portfile [hunk 7]
```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem    1.0
+PortGroup     conflicts_build 1.0
+PortGroup     active_variants 1.1
```
Comment (reneeotten) [inline science/gds-gui/Portfile:5]: both PGs unused?

File: science/gds-gui/Portfile [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+maintainers   {@emaros ligo.org:ed.maros} openmaintainer
+license       GPL
+
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
```
Comment (reneeotten) [inline science/gds-gui/Portfile:20]: please use `https` if possible

File: science/gds-monitor-apps/Portfile [hunk 9]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem    1.0
+PortGroup     conflicts_build 1.0
+PortGroup     active_variants 1.1
```
Comment (reneeotten) [inline science/gds-monitor-apps/Portfile:5]: both PGs seem unused?

File: science/gds-monitor-apps/Portfile [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+maintainers   {@emaros ligo.org:ed.maros} openmaintainer
+license       GPL
+
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
```
Comment (reneeotten) [inline science/gds-monitor-apps/Portfile:20]: please use `https`

File: science/gds-root-extensions/Portfile [hunk 11]
```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem    1.0
+PortGroup     conflicts_build 1.0
+PortGroup     active_variants 1.1
+PortGroup     compiler_blacklist_versions 1.0
```
Comment (reneeotten) [inline science/gds-root-extensions/Portfile:6]: none of the PGs is used?

File: science/gds-root-extensions/Portfile [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+maintainers   {@emaros ligo.org:ed.maros} openmaintainer
+license       GPL
+
+description   LSC Algorithm Library
+long_description \
+  LIGO Scientific Collaboration Algorithm Library containing core \
+  routines for gravitational wave data analysis.
+
+homepage      https://www.lsc-group.phys.uwm.edu/daswg/projects/lalsuite.html
+master_sites  http://software.ligo.org/lscsoft/source/
```
Comment (reneeotten) [inline science/gds-root-extensions/Portfile:21]: `https` if possible?

File: science/gds/Portfile [hunk 13]
```diff
@@ -1,92 +1,64 @@
...
+homepage      https://git.ligo.org/gds/gds
+master_sites  https://software.igwn.org/lscsoft/source/
 
-patchfiles    patch-std_cplusplus_2017.diff
-
-checksums     rmd160  3b2186057dec5d1e530ee1cefbfd12ab4f976d40 \
-              sha256  e592727d686af1042d775f192531d440600105c12d7cdad63455cfad3fb9913c
+checksums     rmd160  b297ab63927141355fdd3e165b6983c4aac04d1a \
+              sha256  6d53977e7538dcff1cf0f48a56ff7f08c05587b82708f6d3e32989f0b8d1e6dd \
+              size    2425166
```
Comment (reneeotten) [inline science/gds/Portfile:22]: there is a checksum mismatch which causes the build failure - see my [first comment](https://github.com/macports/macports-ports/pull/7498#issuecomment-649494999) in this thread.

Comment (emaros) [reply science/gds/Portfile:22]: I had noticed these on my local build after I had done a clean --all. I believe that build 12439 should have them all resolved.

**Comment by reneeotten** [comment]
please double-check your checksums once more, I think that `gds-dtt`, `gds-root-extensions`, `gds-swig`, and `gds-utilities` are still all wrong. Please make sure you do a `sudo port clean --all <portname>` first before calculating the checksums again.

**Comment by reneeotten** [comment]
ping @emaros - are you willing/able to make the final changes to push this PR over the finish line?

**Comment by reneeotten** [comment]
> ping @emaros - are you willing/able to make the final changes to push this PR over the finish line?

okay, I am working on this PR right now and will push an update once it's finalized.

**Comment by emaros** [comment]
Thank you for working on it. I am currently working on the next software release of this suite. I am incorporating changes here into the upstream code base.

**Comment by reneeotten** [comment]
> Thank you for working on it. I am currently working on the next software release of this suite. I am incorporating changes here into the upstream code base.

I'll push to this PR, but cannot get `gds` to build locally. Perhaps you can take a look at the CI output and figure out what the issue is (or perhaps it's already fixed upstream)?

**Comment by reneeotten** [comment]
@emaros please check the CI log for macos-10.15 ; the error is when building `gds` in the log file around line 950

**Comment by emaros** [comment]
> @emaros please check the CI log for macos-10.15 ; the error is when building `gds` in the log file around line 950

Looking at the location, it is one of the issues I just fixed on the upstream source last night. The software, being as old as it is, has its own thread namespace. With the most recent version of XCode and threads being available through the standard C++ library, instances like what you see near line 950 require ::thread.

**Comment by reneeotten** [comment]
> > @emaros please check the CI log for macos-10.15 ; the error is when building `gds` in the log file around line 950
> 
> Looking at the location, it is one of the issues I just fixed on the upstream source last night. The software, being as old as it is, has its own thread namespace. With the most recent version of XCode and threads being available through the standard C++ library, instances like what you see near line 950 require ::thread.

okay, I updated all ports to their latest versions - let's see if the builds now finish.

**Comment by reneeotten** [comment]
@emaros okay, I updated to the latest versions. Now `gds` builds, but pretty much everything else fails.... I think the Portfiles itself are in good shape now, but I don't have the interest or time to chase down the build failures. Feel free to grab this PR and continue from here.

**Comment by reneeotten** [comment]
@emaros I tried updating to the latest versions, but it still doesn't build... the issue appears to be with `gds`. All the logs are available from the CI runs if you want to try and resolve them upstream. I have no intentions to work on this any further.

**Comment by reneeotten** [comment]
@emaros with the last changes/updates I attempted, `gds` did build correctly, but now other ports failed with missing packages. Can't remember the exact details anymore, but perhaps you guys split out more stuff? In any case, feel free to re-open this PR when upstream has has resolved the build failures and/or you are willing/able to fix it in MacPorts.

**Pull request closed without merging**